pkg/instrument/dqlite.go

```diff
@@ -0,0 +1,125 @@
+package instrument
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"sync"
+	"sync/atomic"
+)
+
+type Listener struct {
+	Network, Address        string
+	AcceptedChan            chan net.Conn
+	netListener             net.Listener
+	bytesWritten, bytesRead atomic.Int64
+	wg                      sync.WaitGroup
+	err                     error
+}
+
+type ListenerMetrics struct {
+	BytesWritten int64
+	BytesRead    int64
+}
+
+func NewListener(network, address string) *Listener {
+	return &Listener{
+		Network:      network,
+		Address:      address,
+		AcceptedChan: make(chan net.Conn),
+		netListener:  nil,
```

Comment:
As other fields are taking their zero values, I think it's okay to omit this `nil`-assignment here. I think the absence of `netListener` in `NewListener` is a sufficient indicator that this field is nillable